lib/ansible/modules/database/postgresql/postgresql_facts.py

```diff
@@ -0,0 +1,878 @@
+            via a Unix socket on the server machine.
+          returned: always
+          type: str
+          sample: 10.0.0.101
+        client_hostname:
+          description:
+          - Host name of the connected client, as reported by a reverse DNS lookup of client_addr.
+          - This field will only be non-null for IP connections, and only when log_hostname is enabled.
+          returned: always
+          type: str
+          sample: dbsrv1
+        backend_start:
+          description: Time when this process was started, i.e., when the client connected to this WAL sender.
+          returned: always
+          type: str
+          sample: "2019-02-03 00:14:33.908593+03"
+        state:
+          description: Current WAL sender state.
+          returned: always
+          type: str
+          sample: streaming
+    tablespaces:
+      description:
+      - Information about tablespaces
+        U(https://www.postgresql.org/docs/current/catalog-pg-tablespace.html).
+      returned: always
+      type: dict
+      sample: test
+      contains:
+        spcacl:
+          description: Tablespace access privileges.
+          returned: always
+          type: str
+          sample: "{postgres=C/postgres,andreyk=C/postgres}"
+        spcoptions:
+          description: Tablespace-level options.
+          returned: always
+          type: list
+          sample: [ "seq_page_cost=1" ]
+        spcowner:
+          description: Owner of the tablespace.
+          returned: always
+          type: str
+          sample: test_user
+    roles:
+      description:
+      - Information about roles U(https://www.postgresql.org/docs/current/user-manag.html).
+      returned: always
+      type: dict
+      sample: test_role
+      contains:
+        canlogin:
+          description: Login privilege U(https://www.postgresql.org/docs/current/role-attributes.html).
+          returned: always
+          type: bool
+          sample: true
+        member_of:
+          description: Role membership U(https://www.postgresql.org/docs/current/role-membership.html).
+          returned: always
+          type: list
+          sample: [ "read_only_users" ]
+        superuser:
+          description: User is a superuser or not.
+          returned: always
+          type: bool
+          sample: false
+        valid_until:
+          description: Password expiration date U(https://www.postgresql.org/docs/current/sql-alterrole.html).
+          returned: always
+          type: str
+          sample: "9999-12-31T23:59:59.999999+00:00"
+    settings:
+      description:
+      - Information about run-time server parameters
+        U(https://www.postgresql.org/docs/current/view-pg-settings.html).
+      returned: always
+      type: dict
+      sample: work_mem
+      contains:
+        setting:
+          description: Current value of the parameter.
+          returned: always
+          type: str
+          sample: 49152
```

Comment:
We prefer that returned information is in bytes here. So that values can be converted without complexity.

Reply:
@dagwieers I'm afraid the implementation of this suggestion makes the code very complex because a parameter can have a value like in following output from PostgreSQL system view pg_settings:
```
                name                 |                setting                 | unit 
-------------------------------------+----------------------------------------+------
 allow_system_table_mods             | off                                    | 
 application_name                    | psql                                   | 
 archive_command                     | (disabled)                             | 
 archive_mode                        | off                                    | 
 archive_timeout                     | 0                                      | s
 array_nulls                         | on                                     | 
 authentication_timeout              | 60                                     | s
 autovacuum                          | on                                     | 
 autovacuum_analyze_scale_factor     | 0.1                                    | 
 autovacuum_analyze_threshold        | 50                                     | 
 autovacuum_freeze_max_age           | 200000000                              | 
 autovacuum_max_workers              | 3                                      | 
 autovacuum_multixact_freeze_max_age | 400000000                              | 
 autovacuum_naptime                  | 60                                     | s
 autovacuum_vacuum_cost_delay        | 20                                     | ms
 autovacuum_vacuum_cost_limit        | -1                                     | 
 autovacuum_vacuum_scale_factor      | 0.2                                    | 
 autovacuum_vacuum_threshold         | 50                                     | 
 autovacuum_work_mem                 | -1                                     | kB
 backend_flush_after                 | 0                                      | 8kB
 backslash_quote                     | safe_encoding                          | 
 bgwriter_delay                      | 200                                    | ms
 bgwriter_flush_after                | 64                                     | 8kB
 bgwriter_lru_maxpages               | 100                                    | 
 bgwriter_lru_multiplier             | 2                                      | 
 block_size                          | 8192                                   | 
 bonjour                             | off                                    | 
 bonjour_name                        |                                        | 
 bytea_output                        | hex                                    | 
 check_function_bodies               | on                                     | 
 checkpoint_completion_target        | 0.5                                    | 
 checkpoint_flush_after              | 32                                     | 8kB
 checkpoint_timeout                  | 300                                    | s
 checkpoint_warning                  | 30                                     | s
 client_encoding                     | UTF8                                   | 
 client_min_messages                 | notice                                 | 
 cluster_name                        |                                        | 
 commit_delay                        | 0                                      | 
 commit_siblings                     | 5                                      | 
 config_file                         | /var/lib/pgsql/10/data/postgresql.conf | 
```

Reply:
Remember that you won't be able to change this in the future, because people depend on this design.

Reply:
I added an additional value val_in_bytes that will be returned if supported:
```
        "bgwriter_flush_after": {
            "boot_val": "64", 
            "context": "sighup", 
            "max_val": "256", 
            "min_val": "0", 
            "setting": "64", 
            "sourcefile": "", 
            "unit": "8kB", 
            "val_in_bytes": 524288, 
            "vartype": "integer"
        }, 
```